src/utils/build_importer.py

```diff
@@ -0,0 +1,116 @@
+from logger import Logger
+from pathlib import Path
+import os
+import re
+from selenium import webdriver
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from yaml import dump
+
+
+def import_build():
+    Logger.info(f"Paste maxroll.gg build guide or planner build here ie https://maxroll.gg/d4/build-guides/minion-necromancer-guide")
+    url = input()
+    if "maxroll.gg/d4/planner" not in url and "maxroll.gg/d4/build-guides/" not in url:
+        Logger.info(f"Invalid url, please use a d4 planner build on MaxRoll.gg")
+        return
+    options = webdriver.ChromeOptions()
+    options.add_argument("--headless=new")
+    driver = webdriver.Chrome(options=options)
+    driver.get(url)
+    wait = WebDriverWait(driver, 10)
+    wait.until(EC.element_to_be_clickable((By.CLASS_NAME, "ncmp__btn")))
+    driver.find_elements(By.CLASS_NAME, "ncmp__btn")[1].click()
+    if "maxroll.gg/d4/build-guides/" in url:
+        original_window = driver.current_window_handle
+        driver.find_element(By.CLASS_NAME, "d4t-PlannerLink").click()
+        wait.until(EC.number_of_windows_to_be(2))
+        for window_handle in driver.window_handles:
+            if window_handle != original_window:
+                driver.switch_to.window(window_handle)
+                break
+    planner = wait.until(EC.element_to_be_clickable((By.CLASS_NAME, "d4t-Paperdoll")))
+    item_slots = planner.find_elements(By.CLASS_NAME, "d4t-slot")
+    item_dict = {}
+    item_dict["class"] = driver.find_element(By.CLASS_NAME, "d4t-ClassSelect").text
+    filter_label = driver.find_element(By.CLASS_NAME, "d4t-label").text
+    Logger.info(f"Importing {item_dict["class"]} {filter_label} build")
+
+    for item in item_slots:
+        item.click()
+        driver.find_element(By.CLASS_NAME, "d4t-title").click()
+        item_type = wait.until(EC.element_to_be_clickable((By.CLASS_NAME, "d4t-category-header"))).text
+        if item_type == "Weapon" or item_type == "Equipment":
+            # if this is just "Weapon" then the planner uses different builds at different stages of the build, so we find the active weapon for this current version of the build, and grab the name of the weapon type
+            item_type = (
+                driver.find_element(By.CLASS_NAME, "d4t-ItemLibrary")
```

Comment:
Instead of this complex nested query, you could just provide one xpath, no?

Reply:
Potentially, I've not really used XPath before, I did give it a try but couldn't get it to work so just did it this way. Not ideal, but at least it's clear what it's doing and doesn't need to be performant here.